alicloud/data_source_alicloud_lindorm_instances_test.go

```diff
@@ -0,0 +1,141 @@
+		}),
+	}
+	nameRegexConf := dataSourceTestAccConfig{
+		existConfig: testAccCheckAlicloudLindormInstancesDataSourceName(rand, map[string]string{
+			"name_regex": `"${alicloud_lindorm_instance.default.instance_name}"`,
+		}),
+		fakeConfig: testAccCheckAlicloudLindormInstancesDataSourceName(rand, map[string]string{
+			"name_regex": `"${alicloud_lindorm_instance.default.instance_name}_fake"`,
+		}),
+	}
+	statusConf := dataSourceTestAccConfig{
+		existConfig: testAccCheckAlicloudLindormInstancesDataSourceName(rand, map[string]string{
+			"ids":    `["${alicloud_lindorm_instance.default.id}"]`,
+			"status": `"ACTIVATION"`,
+		}),
+		fakeConfig: testAccCheckAlicloudLindormInstancesDataSourceName(rand, map[string]string{
+			"ids":    `["${alicloud_lindorm_instance.default.id}"]`,
+			"status": `"CREATING"`,
+		}),
+	}
+	allConf := dataSourceTestAccConfig{
+		existConfig: testAccCheckAlicloudLindormInstancesDataSourceName(rand, map[string]string{
+			"ids":            `["${alicloud_lindorm_instance.default.id}"]`,
+			"name_regex":     `"${alicloud_lindorm_instance.default.instance_name}"`,
+			"status":         `"ACTIVATION"`,
+			"support_engine": `"4"`,
+			"query_str":      `"${alicloud_lindorm_instance.default.instance_name}"`,
+		}),
+		fakeConfig: testAccCheckAlicloudLindormInstancesDataSourceName(rand, map[string]string{
+			"ids":            `["${alicloud_lindorm_instance.default.id}_fake"]`,
+			"name_regex":     `"${alicloud_lindorm_instance.default.instance_name}_fake"`,
+			"status":         `"CREATING"`,
+			"support_engine": `"4"`,
+			"query_str":      `"${alicloud_lindorm_instance.default.instance_name}_fake"`,
+		}),
+	}
+	var existAlicloudLindormInstancesDataSourceNameMapFunc = func(rand int) map[string]string {
+		return map[string]string{
+			"ids.#":                     "1",
+			"instances.#":               "1",
+			"instances.0.instance_name": fmt.Sprintf("tf-testAccLindormInstances-%d", rand),
+		}
+	}
+	var fakeAlicloudLindormInstancesDataSourceNameMapFunc = func(rand int) map[string]string {
+		return map[string]string{
+			"ids.#":       "0",
+			"instances.#": "0",
+		}
+	}
+
+	var alicloudLindormInstancesCheckInfo = dataSourceAttr{
+		resourceId:   "data.alicloud_lindorm_instances.default",
+		existMapFunc: existAlicloudLindormInstancesDataSourceNameMapFunc,
+		fakeMapFunc:  fakeAlicloudLindormInstancesDataSourceNameMapFunc,
+	}
+	alicloudLindormInstancesCheckInfo.dataSourceTestCheck(t, rand, idsConf, statusConf, nameRegexConf, supportEngineStrConf, queryStrConf, allConf)
+}
+func testAccCheckAlicloudLindormInstancesDataSourceName(rand int, attrMap map[string]string) string {
+	var pairs []string
+	for k, v := range attrMap {
+		pairs = append(pairs, k+" = "+v)
+	}
+
+	config := fmt.Sprintf(`
+variable "name" {	
+	default = "tf-testAccLindormInstances-%d"
+}
+provider "alicloud" {
+  region = "cn-shenzhen"
+}
+locals {
+  zone_id = "cn-shenzhen-e"
+}
+data "alicloud_vpcs" "default" {
```

Comment:
Please remove the provider config in the testcase template and the region should be set by environment ALICLOUD_REGION